league_rpc_linux/processes/lcu_thread.py

```diff
@@ -0,0 +1,355 @@
+import time
+import lcu_driver
+from league_rpc_linux.colors import Colors
+from threading import Timer
+
+from league_rpc_linux.const import (
+    GAME_MODE_CONVERT_MAP,
+    LEAGUE_OF_LEGENDS_LOGO,
+    MAP_ICON_CONVERT_MAP,
+    BASE_MAP_ICON_URL,
+    PROFILE_ICON_BASE_URL,
+)
+
+connector = lcu_driver.Connector()
+
+rpc = None
+
+data = {
+    "champion_name": None,
+    "skin_name": None,
+    "skin_id": None,
+    "level": None,
+    "gamemode": None,
+    "isCustom": False,
+    "isPractice": False,
+    "inLobby": False,
+    "inQueue": False,
+    "inChampSelect": False,
+    "max_players": None,
+    "players": None,
+    "summoner_name": None,
+    "summoner_id": None,
+    "summoner_icon": None,
+    "summoner_rank": None,
+    "queue": None,
+    "queue_id": -1,
+    "map_id": None,  # 11, 12, 21, 22, 30
+    "lobbyQueueIsRanked": False,  # True, False
+    "lobby_id": None,
+    "availability": "Idle",  # Idle, Away
+    "gameflowPhase": None,  # Lobby, Matchmaking, ReadyCheck, ChampSelect, InProgress, WaitingForStats, EndOfGame
+}
+
+## Timers ##
+
+scheduledUpdate = False
+
+
+# As some events are called multiple times, we should limit the amount of updates to the RPC.
+# Collect update events for 1 second and then update the RPC.
+def delayUpdate():
+    global scheduledUpdate
+
+    if not scheduledUpdate:
+        scheduledUpdate = True
+        Timer(1.0, update_rpc_and_reset_flag).start()
+
+
+def update_rpc_and_reset_flag():
```

Comment:
I'm having a hard time understanding why we need this. what does this achieve?

Reply:
I dont want to spam RPC with multiple updates per second in case the websocket update functions are triggered multiple times. 
Like /lobby is triggered really often in short succession ~1sec = 4-5 calls.